docs/checklist-command.md

```diff
@@ -0,0 +1,32 @@
+# New CLI command
+
+Subcommands should always be preferred over adding more top-level commands.
+
+Code flow for commands is generally:
+
+```
+CLI (command/) -> API Client (api/) -> HTTP API (command/agent) -> RPC (nomad/)
+```
+
+## Code
+
+1.  [ ] New file in `command/` or in an existing file if a subcommand
+2.  [ ] Test new command in `command/` package
+3.  [ ] Implement autocomplete
+4.  [ ] Implement `-json` (returns raw API response)
+5.  [ ] Implement `-verbose` (expands truncated UUIDs, adds other detail)
+7.  [ ] Update help text
+7.  [ ] Implement and test new HTTP endpoint in `command/agent/<command>_endpoint.go`
+8.  [ ] Implement and test new RPC endpoint in `nomad/<command>_endpoint.go`
+9.  [ ] Implement and test new Client RPC endpoint in
+    `client/<command>_endpoint.go` (For client endpoints like Filesystem only)
+10. [ ] Implement and test new `api/` package Request and Response structs
+11. [ ] Implement and test new `api/` package helper methods
+12. [ ] Implement and test new `nomad/structs/` package Request and Response structs
```

Comment:
If we want to number the list, it might be better to reorder them in terms of order of implementation (e.g. structs need to be added before consumers), and I may suggest chunking them (e.g. RPC Handlers, HTTP Handlers, CLI command).

Reply:
Also numbers are not rendered here, it seems checklist takes precedence over ordered list in markdown. If we want explicit ordering, we should move the numbering inside the checklist. (The plus side of this approach is that nested checklists and unordered lists are supported under checklists, which could make the Step 9 sub-bullet-point render nicely.)

Reply:
Ha, totally didn't notice the numbers disappeared. Switched to bullet points as the order doesn't really make sense anyway.

I'm torn on how useful ordering would be. If somebody wants to sketch out a command via the help text in the command package first, that seems as reasonable as starting at the inside of the onion (structs) and working your way out.

I'd rather not block this on organizational tweaks. We can followup with that if anyone finds a particularly compelling order.